coding.go

```diff
@@ -1,62 +1,18 @@
 package format
 
 import (
-	"fmt"
-	"sync"
-
 	blocks "github.com/ipfs/go-block-format"
 )
 
 // DecodeBlockFunc functions decode blocks into nodes.
 type DecodeBlockFunc func(block blocks.Block) (Node, error)
 
-type BlockDecoder interface {
-	Register(codec uint64, decoder DecodeBlockFunc)
-	Decode(blocks.Block) (Node, error)
-}
-type safeBlockDecoder struct {
-	// Can be replaced with an RCU if necessary.
-	lock     sync.RWMutex
-	decoders map[uint64]DecodeBlockFunc
-}
-
-// Register registers decoder for all blocks with the passed codec.
-//
-// This will silently replace any existing registered block decoders.
-func (d *safeBlockDecoder) Register(codec uint64, decoder DecodeBlockFunc) {
-	d.lock.Lock()
-	defer d.lock.Unlock()
-	d.decoders[codec] = decoder
-}
-
-func (d *safeBlockDecoder) Decode(block blocks.Block) (Node, error) {
```

Comment:
Any preferences on whether removing this is a good/bad idea? We could just expose this as a registry any application can implement (as in the go-ipld-legacy PR).

Reply:
@Jorropo @willscott @rvagg I'm thinking of not keeping the interface, but exposing a non-threadsafe registry here like exists in go-ipld-prime. This makes upgrading easier for people without having to consider implications like "should I use the go-ipld-format codec or the go-ipld-prime version of the same one". People can then opt into the go-ipld-prime codecs optionally/as needed (e.g. since there are more of them around these days).